### PR TITLE
Added example in the doc for TypeVar in using generic classes

### LINKED ISSUE #8523: Error when preserving generic type behaviour in subclasses (when not supported at runtime)
This is a bug report.
* mypy = 0.770
* python = 3.8.2
* I did not try running mypy from git master
----
The docs for "Using classes that are generic in stubs but not at runtime" ([here](https://mypy.readthedocs.io/en/stable/common_issues.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime)) clearly indicate how to subclass classes that are defined to be generic in their stubs but don't actually support generics in their runtime code.

The shown method (conditionally defining a typed subclass) works great when you want to lock the type down in your subclass. However, **if you want to maintain the subclass as a generic it does not work** (mypy reports an error).

Here is failing case (with `mypy --strict`), where the only variation from the canonical [example](https://mypy.readthedocs.io/en/stable/common_issues.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime) is using a generic `TypeVar` instead of a `str` as the type for the Queue:

```python
from queue import Queue
from typing import TYPE_CHECKING, TypeVar

_T = TypeVar("_T")
if TYPE_CHECKING:
    MyQueueBase = Queue[_T]  # change _T to str here and it works perfectly (per the docs)
else:
    MyQueueBase = Queue
class MyQueue(MyQueueBase): pass  # mypy error is on this line (9)
```

mypy reports an error as follows (when the base is `Queue[_T]`):

```
$ python --version
Python 3.8.2
$ mypy --version
mypy 0.770
$ mypy --strict ./generic_subclass_example.py 
generic_subclass_example.py:9: error: Missing type parameters for generic type "MyQueueBase"
Found 1 error in 1 file (checked 1 source file)
```

Note that without `--strict` there is no error (not sure which of the flags causes it).

### PR DESCRIPTION
**Pull request by ashutosh1919**
Fixes #8523 . 
@JukkaL and @ethanhs, Please review this PR.

### CODE REVIEW

File: docs/source/common_issues.rst [hunk 1]
```diff
@@ -595,6 +595,40 @@ string literal types or :py:data:`~typing.TYPE_CHECKING`:
...
+   from queue import Queue
+   from typing import TYPE_CHECKING, TypeVar
+
+   _T = TypeVar("_T")
+   if TYPE_CHECKING:
+       MyQueueBase = Queue[_T]
+   else:
+       MyQueueBase = Queue
+
+   class MyQueue(MyQueueBase): pass  # error: Missing type parameters for generic type
```
Comment (JukkaL) [inline docs/source/common_issues.rst]: As mentioned above, the error only comes when some configuration options are enabled.

File: docs/source/common_issues.rst [hunk 2]
```diff
@@ -595,6 +595,40 @@ string literal types or :py:data:`~typing.TYPE_CHECKING`:
 
    results: 'Queue[int]' = Queue()  # OK
 
+But while using :py:data:`~typing.TypeVar`, observe here that it requires parameters
+while using it as a type. And if we implement it in above way it will give error as
```
[the remark this answers is not in the document]
Comment (ashutosh1919) [reply docs/source/common_issues.rst]: done

File: docs/source/common_issues.rst (hunk 1 above)
Comment (ashutosh1919) [reply docs/source/common_issues.rst:616]: done

Comment (ashutosh1919) [reply docs/source/common_issues.rst:616]: done

File: docs/source/common_issues.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashutosh1919) [reply docs/source/common_issues.rst]: done

**Comment by sbdchd** [comment]
Any update on this? Took me some spelunking to find it

**Comment by hauntsaninja** [comment]
Subsumed by https://github.com/python/mypy/pull/9763

**Pull request closed without merging**